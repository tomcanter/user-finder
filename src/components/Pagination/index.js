// @flow

import React, {
  PropTypes,
} from 'react';
import {Link} from 'react-router-dom';
import first from 'lodash/fp/first';
import flow from 'lodash/fp/flow';

type Props = {
  next: Object,
  prev: Object,
};

const getSearchQuery = flow(
  url => url.match(/(\?.+)/),
  first
);

function renderLink(url: string, text: string) {
  if (!url) {return null;}
  const search = getSearchQuery(url);
  return <Link to={{search}}>{text}</Link>;
}

function Pagination({prev, next}: Props) {
  return (
    <div>
      {renderLink(prev.url, 'Previous')}
      {renderLink(next.url, 'Next')}
    </div>
  );
}

Pagination.defaultProps = {
  next: {},
  prev: {},
};

Pagination.propTypes = {
  next: PropTypes.object,
  prev: PropTypes.object,
};

export default Pagination;
