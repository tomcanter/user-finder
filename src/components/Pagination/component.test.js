import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrowserRouter as Router,
} from 'react-router-dom';

import Pagination from './';

function createComponent(props) {
  return renderer.create(
    <Router>
      <Pagination {...props} />
    </Router>
  );
}

describe('Component: Pagination', () => {

  describe('when passed a next prop', () => {
    it('should render a next link', () => {
      const props = {
        next: {
          url: 'https://api.github.com/search/users?q=tom&page=2',
        },
      };
      const component = createComponent(props);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when passed a prev prop', () => {
    it('should render a previous link', () => {
      const props = {
        prev: {
          url: 'https://api.github.com/search/users?q=tom&page=1',
        },
      };
      const component = createComponent(props);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when passed a prev and next prop', () => {
    it('should render a previous and next link', () => {
      const props = {
        prev: {
          url: 'https://api.github.com/search/users?q=tom&page=1',
        },
        next: {
          url: 'https://api.github.com/search/users?q=tom&page=2',
        },
      };
      const component = createComponent(props);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

});
