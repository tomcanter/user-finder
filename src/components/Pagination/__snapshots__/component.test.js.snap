exports[`Component: Pagination when passed a next prop should render a next link 1`] = `
<div
  className="">
  <div
    className="Container_inner_txqaf5">
    <div
      className="Pagination_2gx8tz">
      <a
        className="Pagination_next_1j0ix3o"
        href="/?q=tom&page=2"
        onClick={[Function]}>
        Next
      </a>
    </div>
  </div>
</div>
`;

exports[`Component: Pagination when passed a prev and next prop should render a previous and next link 1`] = `
<div
  className="">
  <div
    className="Container_inner_txqaf5">
    <div
      className="Pagination_2gx8tz">
      <a
        className="Pagination_prev_yunyae"
        href="/?q=tom&page=1"
        onClick={[Function]}>
        Previous
      </a>
      <a
        className="Pagination_next_1j0ix3o"
        href="/?q=tom&page=2"
        onClick={[Function]}>
        Next
      </a>
    </div>
  </div>
</div>
`;

exports[`Component: Pagination when passed a prev prop should render a previous link 1`] = `
<div
  className="">
  <div
    className="Container_inner_txqaf5">
    <div
      className="Pagination_2gx8tz">
      <a
        className="Pagination_prev_yunyae"
        href="/?q=tom&page=1"
        onClick={[Function]}>
        Previous
      </a>
    </div>
  </div>
</div>
`;
